pymc3/distributions/discrete.py

```diff
@@ -689,9 +697,18 @@ def random(self, point=None, size=None, repeat=None):
         return reshape_sampled(sampled, size, self.shape)
 
     def logp(self, value):
-        return tt.switch(value > 0,
-                         tt.log(self.psi) + self.bin.logp(value),
-                         tt.log((1. - self.psi) + self.psi * tt.pow(1 - self.p, self.n)))
+        psi = self.psi
+        p = self.p
+        n = self.n
+
+        logp_val = tt.switch(value > 0,
+                 logsumexp(tt.log(psi) + self.bin.logp(value)),
```

Comment:
The exp is missing :-)
It helps if you have an expression like log(exp(a) + exp(b)). But we are only doing log(a + b) here.